src/Eccube/Form/Type/Admin/MasterdataDataType.php

```diff
@@ -53,15 +55,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'max' => $app['config']['int_len'],
                     )),
                     new Assert\Regex(array(
-                        'pattern' => "/^\d+$/u",
-                        'message' => 'form.type.numeric.invalid',
+                        'pattern' => '/^\d+$/u',
+                        'message' => $app->trans('form.type.numeric.invalid'),
                     )),
                 ),
             ))
             ->add('name', 'text', array(
                 'required' => false,
             ))
-            ;
+        ->addEventListener(FormEvents::POST_SUBMIT, function (FormEvent $event) use ($app) {
+            $form = $event->getForm();
+            $data = $form->getData();
+            if (strlen($data['id']) && strlen($data['name']) == 0) {
+                $form['name']->addError(new FormError($app->trans('This value should not be blank.')));
```

Comment:
`$app->trans` is not need. (please, delete.)

ref:
dump($app->trans('This value should not be blank.'));
=> `"This value should not be blank."` (its meaningless)